src/NuGet.Services.AzureSearch/Owners2AzureSearch/OwnerSetComparer.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+
+namespace NuGet.Services.AzureSearch.Owners2AzureSearch
+{
+    public class OwnerSetComparer : IOwnerSetComparer
+    {
+        private static readonly string[] EmptyStringArray = new string[0];
+
+        private readonly ILogger _logger;
+
+        public OwnerSetComparer(ILogger logger)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public SortedDictionary<string, string[]> Compare(
```

Comment:
>Compare [](start = 50, length = 7)

Should we assert that the `oldData` and `newData` have the expected `.Comparer`?

Reply:
Certainly doesn't hurt.